mmv1/templates/tgc_next/tfplan2cai/expand_array_resourceref_with_validation.go.tmpl

```diff
@@ -0,0 +1,34 @@
+{{/*
+	The license inside this block applies to this file
+	Copyright 2025 Google Inc.
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/ -}}
+{{- define "expandArrayResourcerefWithValidation" }}
+func expand{{$.GetPrefix}}{{$.TitlelizeProperty}}(v interface{}, d tpgresource.TerraformResourceData, config *transport_tpg.Config) (interface{}, error) {
+{{- if $.IsSet }}
+  v = v.(*schema.Set).List()
+{{- end }}
+  l := v.([]interface{})
+  req := make([]interface{}, 0, len(l))
+  for _, raw := range l {
+    if raw == nil {
+      return nil, fmt.Errorf("Invalid value for {{underscore $.Name}}: nil")
+    }
+    f, err := {{ template "expandResourceRef" dict "VarName" "raw.(string)" "ResourceRef" $.ItemType.ResourceRef "ResourceType" $.ItemType.ResourceType}}
+    if err != nil {
+      return nil, fmt.Errorf("Invalid value for {{underscore $.Name}}: %s", err)
+    }
```

Comment:
Should return an error if v is not a slice.

Also, can we rename `l` to `vSlice`? `l` is hard to read.

Reply:
It is safe to add the check.